dockstore-webservice/src/main/java/io/dockstore/webservice/core/Doi.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Map;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+@Entity
+@Schema(description = "A Digital Object Identifier (DOI)")
+@Table(name = "doi", uniqueConstraints = @UniqueConstraint(name = "unique_doi_name", columnNames = { "name" }))
+public class Doi {
+    // DOI order of precedence from greatest to least
+    public static final List<DoiCreator> DOI_ORDER_OF_PRECEDENCE = List.of(DoiCreator.USER, DoiCreator.GITHUB, DoiCreator.DOCKSTORE);
+    public static final int MAX_NUMBER_OF_DOI_CREATORS = 3;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Schema(description = "Implementation specific ID for the DOI in this web service")
+    private long id;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.STRING)
+    @Schema(description = "The type of DOI", requiredMode = RequiredMode.REQUIRED, example = "CONCEPT")
+    private DoiType type;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.STRING)
+    @Schema(description = "The DOI creator", requiredMode = RequiredMode.REQUIRED, example = "USER")
+    private DoiCreator creator;
+
+    @Column(nullable = false)
+    @Schema(description = "The DOI name", requiredMode = RequiredMode.REQUIRED, example = "10.5281/zenodo.705645")
+    private String name;
+
+    @Column
+    @Schema(description = "The ID of the access link with edit permissions")
+    private String editAccessLinkId;
+
+    // database timestamps
+    @Column(updatable = false)
+    @CreationTimestamp
+    private Timestamp dbCreateDate;
+
+    @Column()
+    @UpdateTimestamp
+    private Timestamp dbUpdateDate;
+
+    public Doi() {
+    }
+
+    public Doi(DoiType type, DoiCreator creator, String name) {
+        this.type = type;
+        this.creator = creator;
+        this.name = name;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public DoiType getType() {
+        return type;
+    }
+
+    public void setType(DoiType type) {
+        this.type = type;
+    }
+
+    public DoiCreator getCreator() {
+        return creator;
+    }
+
+    public void setCreator(DoiCreator creator) {
+        this.creator = creator;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getEditAccessLinkId() {
+        return editAccessLinkId;
+    }
+
+    public void setEditAccessLinkId(String editAccessLinkId) {
+        this.editAccessLinkId = editAccessLinkId;
+    }
+
+    public static Doi getDoiBasedOnOrderOfPrecedence(Map<DoiCreator, Doi> dois) {
+        for (DoiCreator doiCreator: DOI_ORDER_OF_PRECEDENCE) {
+            if (dois.containsKey(doiCreator)) {
+                return dois.get(doiCreator);
+            }
+        }
+        return null;
+    }
+
+    public enum DoiType {
+        CONCEPT,
+        VERSION
```

Comment:
If I understand correctly, "concept" and "version" are Zenodo concepts.  The DOI standard has no such abstractions, you can point a DOI at whatever you want.   No change necessary here.